static/js/settings_org.js

```diff
@@ -419,8 +419,29 @@ function _set_up() {
     });
 
     function property_value_element_refers(property_name) {
+        var current_val;
         if (property_name === 'realm_message_content_edit_limit_minutes') {
             return Math.ceil(page_params.realm_message_content_edit_limit_seconds / 60).toString();
+        } else if (property_name === 'realm_create_stream_permission') {
+            if (page_params.create_stream_by_admins_only) {
+                current_val = "by_admins_only";
+            } else {
+                if (page_params.realm_waiting_period_threshold === 0) {
+                    current_val = "by_anyone";
+                } else if (page_params.realm_waiting_period_threshold === 3) {
+                    current_val = "by_admin_user_with_three_days_old";
+                } else {
+                    current_val = "by_admin_user_with_custom_time";
```

Comment:
I would make these code paths return.  It'll read better without `current_val`, since you never set its value twice anyway.

Reply:
I was thinking that having so many return statement makes code ugly :)
I'll update it.